packages/store/src/patterns/patternMatchers.js

```diff
@@ -1317,133 +1332,314 @@ const makePatternKit = () => {
       check(false, X`CopyMap not yet supported as keys`),
   });
 
+  /**
+   * @param {Passable[]} specimen
+   * @param {Pattern[]} requiredPatt
+   * @param {Pattern[]} optionalPatt
+   * @returns {{
+   *   requiredSpecimen: Passable[],
+   *   optionalSpecimen: Passable[],
+   *   restSpecimen: Passable[]
+   * }}
+   */
+  const splitArrayParts = (specimen, requiredPatt, optionalPatt) => {
+    const numRequired = requiredPatt.length;
+    const numOptional = optionalPatt.length;
+    const requiredSpecimen = specimen.slice(0, numRequired);
+    const optionalSpecimen = specimen.slice(
+      numRequired,
+      numRequired + numOptional,
+    );
+    const restSpecimen = specimen.slice(numRequired + numOptional);
+    return harden({ requiredSpecimen, optionalSpecimen, restSpecimen });
+  };
+
+  /**
+   * Optional specimen elements which are `undefined` pass unconditionally.
+   * We encode this with the `M.or` pattern so it also produces a good
+   * compression distinguishing `undefined` from absence.
+   *
+   * @param {Pattern[]} optionalPatt
+   * @param {number} length
+   * @returns {Pattern[]} The partialPatt
+   */
+  const adaptArrayPattern = (optionalPatt, length) =>
+    harden(optionalPatt.slice(0, length).map(patt => MM.opt(patt)));
+
   /** @type {MatchHelper} */
```

Comment:
Looks like you've added two methods to MatchHelper but the doc hasn't been updated. Please document what `compress` and `decompress` are for, and whether they're optional on the type or not. There are at least 24 uses of the type and it looks like only this one has these new methods defined.

Reply:
https://github.com/Agoric/agoric-sdk/pull/6597#discussion_r1034784458 `compress` and `decompress` were dead in this PR and are now removed. Thanks for catching! Resolving.